lib/builder.ts

```diff
@@ -98,74 +104,67 @@ class Builder extends EventEmitter {
     }
   }
 
+  build() {
+    // return this.buildPromise();
+    if (this._cancelationRequest) {
+      return Promise.reject(new BuilderError('Cannot start a build if one is already running'));
+    }
+    let promise = this._buildAsync();
+    this._cancelationRequest = new CancelationRequest(promise);
```

Comment:
I believe in the previous code, most failures within the now `buildAsync` would not cause the `this.cancellationRequest` from being re-set. Today, any failure in that could would prevent this.

Is that intentional?

Reply:
I find the way the is written now to be more straightforward. It runs the build. While the build is running, you can cancel it and you can't start a new built. I don't know why we want to allow exceptions to that policy for "most failures" that could occur within the buildAsync method.